### PR TITLE
Minor improvements in handling

### PR DESCRIPTION
**Pull request by XeBoris**
- I started to add some helpful http error messages for failing requests
  - Init & run number / run name selection is separated from each other
  - Interlock run name and run number (both are valid parameters after a cross check with the runDB
  - Focus on the data field: added a replace_data function to allow user to exchange a certain dictionary in the data 'list' by a new one

### CODE REVIEW

**Comment by ershockley** [comment]
I can see how this functionality is useful in your example, but I would like to keep the module useful also for analysis-level queries which do not update the runDB. Analysts might want to use this for finding the run numbers they should use for their analyses, etc (right now I do this with my own code). If we design the DB class to only handle queries involving individual runs then we lose this functionality or at least make it much more difficult. 

Do you see an easy way to do something like the block below with your PR? I know this isn't implemented in the master yet but I was planning to add things like this. Also for things like processing we will want to run queries to see which data needs to be processed. All of these things involve working with lists of runs, not individual ones. 

```
db = utilix.DB()
runlist = db.find_runs(tag='_sciencerun1', source='Kr83m')
```

You are right that in your example we would likely have to pass the run number/name as input multiple times, but in my opinion this is preferable to locking us into this specific use case. 

Thoughts?

**Comment by XeBoris** [comment]
As we discussed, I added the functionality to receive data fields directly by run number/name and detector. A new function [_evaluator()] evaluates the information which are handed over. This allows to run two configurations:
```
db = DB()
db.set_run_number(2000)
db_data = db.get_data()
```
or
```
db = DB()
db.get_data(2000)
```
I think this handles both ways to interact with the data fields. Further requests (e.g. source type,...) are still needed.

File: utilix/rundb.py [hunk 1]
```diff
@@ -77,60 +173,147 @@ def write(self):
...
+        # Generalize your number vs. name approach to fix the input
+        self.run_name = run_name
+        self.run_detector = detector
+        self.run_number = self._get_number(self.run_name, detector=detector)
+        self.set_run = True
+        self._setup()
+
+    #Helper:
+    @Responder
+    def _get(self, url):
```
[the remark this answers is not in the document]
Comment (XeBoris) [reply utilix/rundb.py]: The "private" function _get() returns an objects which is defined by the request library and is processed with json.loads(..). I suggest to introduce simply another "public" get() function which does exactly what your propose.

File: utilix/rundb.py [hunk 2]
```diff
@@ -77,60 +173,147 @@ def write(self):
             json.dump(self.json, f)
 
 
-class DB:
+class DB():
     """Wrapper around the RunDB API"""
 
-    def __init__(self, token_path=os.path.join(os.environ['HOME'], ".dbtoken")):
+    def __init__(self):
```
[the remark this answers is not in the document]
Comment (XeBoris) [reply utilix/rundb.py]: indeed! Forgot to move it back.

File: utilix/rundb.py [hunk 3]
```diff
@@ -77,60 +173,147 @@ def write(self):
...
+
+    def _evaluator(self, run, detector):
+        # Evaluate if run is a run number int or string if
+        # set_run_number() or set_run_name() are not used!
+        if type(run) == int:
+            self.set_run_number(run_number=run, detector=detector)
+        elif type(run) == str:
+            self.set_run_name(run_name=run, detector=detector)
+
+    def _get_name(self, number=None, detector='tpc'):
```
[the remark this answers is not in the document]
Comment (XeBoris) [reply utilix/rundb.py]: removed

**Comment by XeBoris** [comment]
I removed all additional variables which hold the run numbers, names and the type (aka selector). We can not evaluate the input anymore to a certain member function is a run number or run name if we do not store these information somewhere. Therefore, I am simply going to replicate all necessary runDB calls so far (e.g. get_data_by_name() vs. get_data_by_number()). In that sense do we need any super classes or similar which complicates also everything.

File: utilix/rundb.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ershockley) [reply utilix/rundb.py]: Ah, I see what you're saying. Is there ever a time we wouldn't want to decode right away? If not then I would just add the json.loads() to this function and make it public in order to minimize number of lines. But what you propose works too.

**Comment by ershockley** [comment]
Thanks Boris! I'm not sure I understand your last point though. We shouldn't need something like `get_data_by_name` etc. if we have a function like what we talked about that can decide based on the type of input passed what query to make. Or am I missing something?

Anyway I'm happy with these changes and will merge. Thanks!

**Pull request merged**